fabric-entity-events-v1/src/main/java/net/fabricmc/fabric/api/entity/event/v1/ServerPlayerEvents.java

```diff
@@ -69,6 +96,19 @@
 		void afterRespawn(ServerPlayerEntity oldPlayer, ServerPlayerEntity newPlayer, boolean alive);
 	}
 
+	@FunctionalInterface
+	public interface CancelDeath {
```

Comment:
Is this the best name? It can be used for more than just cancel'ing death.

I would have 2 events here, an AllowDeath and a BeforeDeath - this would allow mods to cancel death and for others to lisen to it knowing that the player is actually going to die. See the ScreenMouseEvents for a similar example.